src/js/tooltip.js

```diff
@@ -29,7 +30,8 @@ class Tooltip {
 
 		this.targetNode = document.getElementById(this.opts.target);
 		this.target = new Tooltip.Target(this.targetNode);
-		this.tooltipPosition = this.opts.position;
+		this.oGrid = oGrid;
```

Comment:
I think `oGrid` can be used without assigning it as property to `Target`? That way it's not exposed directly to any consuming code.

Reply:
I had to do that to make it testable, I need to stub oGrid's method

Reply:
I assume you mean 'assigning it as property to *Tooltip*' ?

Reply:
Hmm 🤔 maybe create `Tooltip._getCurrentLayout` to wrap the call to grid. You can then stub `Tooltip._getCurrentLayout` and grid won't be exposed directly.

Reply:
good shout, done. I had to make it a static method to pass lint